sharding/utils/marshal.go

```diff
@@ -0,0 +1,185 @@
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+)
+
+var (
+	chunkSize     = int64(32)
+	indicatorSize = int64(1)
+	chunkDataSize = chunkSize - indicatorSize
+)
+
+// ConvertInterface converts inputted interface to the required type of interface, ex: slice.
+func ConvertInterface(arg interface{}, kind reflect.Kind) ([]interface{}, error) {
+	val := reflect.ValueOf(arg)
+	if val.Kind() == kind {
+
+		return val.Interface().([]interface{}), nil
+
+	}
+	err := errors.New("Interface Conversion a failure")
+	return nil, err
+
+}
+
+func convertbyteToInterface(arg []byte) []interface{} {
+	length := int64(len(arg))
+	newtype := make([]interface{}, length)
+	for i, v := range arg {
+		newtype[i] = v
+	}
+
+	return newtype
+}
+
+func interfacetoByte(arg []interface{}) []byte {
+	length := int64(len(arg))
+	newtype := make([]byte, length)
+	for i, v := range arg {
+		newtype[i] = v.(byte)
+	}
+
+	return newtype
+}
+
+// serializeBlob parses the blob and serializes it appropriately.
+func serializeBlob(cb interface{}) ([]byte, error) {
+
+	interfaceblob, err := ConvertInterface(cb, reflect.Slice)
+	if err != nil {
+		return nil, fmt.Errorf("Error: %v", err)
+	}
+	blob := interfacetoByte(interfaceblob)
```

Comment:
Yeah I have been having a  problem with this. If the library is to accept all kinds of data , then `interface{}` should be able to marshal everything from byte,string,int slices to `[]*types.Transaction` . However a type assertion would require that the value stored in `interface{}` actually be a byte slice. So for other data types any type assertion would lead to panic.

Reply:
I think I will just use the unsafe package and type switch to convert the data types to byte slices